cli/swarm.go

```diff
@@ -0,0 +1,111 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/swarm/api/client"
+)
+
+type SwarmBackend struct {
+	indexDirHash string
+	api          *client.Client
+	indexDir     string
+}
+
+func (bzz *SwarmBackend) OpenIndex(bzzuri string, indexDirHash string) (err error) {
+	if indexDirHash != "" {
+		bzz.indexDirHash = indexDirHash
+	} else {
+		bzz.indexDirHash = bzz.getIndexManifestHash()
+	}
+	bzz.indexDir, err = ioutil.TempDir("", "meta-index-proxy")
+	if err != nil {
+		return err
+	}
+	if bzzuri != "" {
+		bzz.api = client.NewClient(bzzuri)
+	} else {
+		bzz.api = client.DefaultClient
+	}
+	_, err = bzz.api.DownloadManifest(bzz.indexDirHash)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO: will match if file name start is unique, and will not return pathname due to bzz bugs
+func (bzz *SwarmBackend) GetIndexFile(path string, mustfind bool) (string, error) {
+	olddir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	err = os.Chdir(bzz.indexDir)
```

Comment:
Why do we need to change directory here?

Reply:
It seems to make the code clearer to do it.